sky/shell/shell.cc

```diff
@@ -249,5 +248,55 @@ void Shell::WaitForPlatformViewsUIThread(
   latch->Signal();
 }
 
+void Shell::RunInPlatformView(uintptr_t view_id,
+                              const char* main_script,
+                              const char* packages_file,
+                              const char* asset_directory,
+                              bool* view_existed) {
+  base::WaitableEvent latch(false, false);
+  DCHECK(view_id != 0);
+  DCHECK(main_script);
+  DCHECK(packages_file);
+  DCHECK(asset_directory);
+  DCHECK(view_existed);
+
+  ui_task_runner()->PostTask(
+      FROM_HERE,
+      base::Bind(&Shell::RunInPlatformViewUIThread,
+                 base::Unretained(this),
+                 view_id,
+                 base::Unretained(main_script),
+                 base::Unretained(packages_file),
+                 base::Unretained(asset_directory),
+                 base::Unretained(view_existed),
+                 base::Unretained(&latch)));
+
+  latch.Wait();
+}
+
+void Shell::RunInPlatformViewUIThread(uintptr_t view_id,
+                                      const char* main_script,
+                                      const char* packages_file,
+                                      const char* asset_directory,
```

Comment:
IMHO, we should use std::string to transport this data across threads.  That way we don't have to worry about leaking the strings.